tests/protocol/test_wire.py

```diff
@@ -1392,13 +1389,82 @@ def test_it_should_compare_goal_states_when_extensions_config_change(self):
 
             self.assertEqual(1, compare_patcher.call_count, "ExtensionsGoalState.compare() should have been called exactly once")
 
+    def test_it_should_keep_track_of_errors_in_vm_settings_requests(self):
+        mock_response = None
+
+        def http_get_handler(url, *_, **__):
+            if self.is_host_plugin_vm_settings_request(url):
+                if isinstance(mock_response, Exception):
+                    # E0702: Raising NoneType while only classes or instances are allowed (raising-bad-type) - Disabled: we never raise None
+                    raise mock_response  # pylint: disable=raising-bad-type
+                return mock_response
+            return None
+
+        with mock_wire_protocol(mockwiredata.DATA_FILE_VM_SETTINGS, http_get_handler=http_get_handler) as protocol:
+            mock_response = MockHttpResponse(httpclient.INTERNAL_SERVER_ERROR)
+            protocol.client.update_goal_state()
+
+            mock_response = MockHttpResponse(httpclient.BAD_REQUEST)
+            protocol.client.update_goal_state()
+            protocol.client.update_goal_state()
+
+            mock_response = IOError("timed out")
+            protocol.client.update_goal_state()
+
+            mock_response = httpclient.HTTPException()
+            protocol.client.update_goal_state()
+            protocol.client.update_goal_state()
+
+            # force the summary by resetting its period and calling update_goal_state
+            with patch("azurelinuxagent.common.protocol.wire.add_event") as add_event:
+                mock_response = None  # stop producing errors
+                protocol.client._vm_settings_error_reporter._next_period = datetime.now()
+                protocol.client.update_goal_state()
```

Comment:
This can't be replaced by calling the private `_reset` function on the error reporter, can it? That would have the benefit of avoiding any potential side effects of the `update_goal_state` call and removing the need for the `+ 2` down on the definition for `expected['requests']`.

Reply:
no, _reset sets the counters to 0; this actually output the summary